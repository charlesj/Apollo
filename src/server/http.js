
export const HttpConstants = {
  BadRequestName: 'BadRequest',
  NotFoundName: 'NotFound',
  NotFoundMessage: 'Resource not found',
  ServerErrorMessage: 'The server encountered an error processing the request',
  ServerErrorName: 'InternalServerError',
  UnauthorizedMessage: 'The user is not authorized to do this',
  UnauthorizedName: 'Unauthorized',
}

export function accepted(res, message) {
  res.status(202).json({ message })
}

export function badRequest(res, errorMessage, validationErrors) {
  res.status(400).json({
    name: HttpConstants.BadRequestName,
    errorMessage,
    validationErrors,
  })
}

export function created(res, entity) {
  res.status(201).json(entity)
}

export function noContent(res) {
  res.status(204).end()
}

export function notFound(res) {
  res.status(404).json({
    message: HttpConstants.NotFoundMessage,
    name: HttpConstants.NotFoundName,
  })
}

export function ok(res, entity) {
  res.status(200).json(entity)
}

export function serverError(res) {
  res.status(500).json({
    message: HttpConstants.ServerErrorMessage,
    name: HttpConstants.ServerErrorName,
  })
}


export function unauthorized(res) {
  res.status(401).json({
    message: HttpConstants.UnauthorizedMessage,
    name: HttpConstants.UnauthorizedName,
  })
}
