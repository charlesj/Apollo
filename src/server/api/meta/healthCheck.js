import { getConnection } from '../../../data/structure/connection'
import { ok } from '../../utils/http'
import { handleError } from '../../handleError'

export const healthCheck = handleError(async (req, res) => {
  const connection = await getConnection()
  const health = {
    db: !!connection,
  }
  ok(res, health)
})
