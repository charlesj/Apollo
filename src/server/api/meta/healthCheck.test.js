jest.mock('../../..//data/structure/connection')
jest.mock('../../http')

import { getConnection } from '../../../data/structure/connection'

import { healthCheck } from './healthCheck'
import { mockResponse, mockRequest } from '../../../tests/expressMocks'
import { ok } from '../../http'

let req = mockRequest()
let res = mockResponse()

describe('healthCheck', () => {
  it('checks for a valid database connection and returns it', async () => {
    const connection = { connected: true }
    getConnection.mockResolvedValue(connection)
    await healthCheck(req, res)
    expect(ok).toHaveBeenCalledTimes(1)
    expect(ok.mock.calls[0][1]).toMatchSnapshot()
  })
})
