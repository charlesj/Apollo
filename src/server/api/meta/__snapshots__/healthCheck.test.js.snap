// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`healthCheck checks for a valid database connection and returns it 1`] = `
Object {
  "db": true,
}
`;
